smoke_tests/sns_subscribe_test.go

```diff
@@ -58,6 +60,55 @@ func Test_Subscribe_json(t *testing.T) {
 	assert.Equal(t, "sqs", subscriptions[0].Protocol)
 	assert.False(t, subscriptions[0].Raw)
 	assert.Contains(t, subscriptions[0].SubscriptionArn, fmt.Sprintf("%s:%s", af.BASE_SNS_ARN, "unit-topic2"))
+	assert.Equal(t, response.SubscriptionArn, &subscriptions[0].SubscriptionArn)
+}
+
+func Test_Subscribe_json_with_duplicate_subscription(t *testing.T) {
```

Comment:
This test asserts that the subscription should not be duplicated when the multiple same subscriptions are requested.